runtimes/go/beta/errs/error.go

```diff
@@ -106,9 +109,13 @@ func WrapCode(err error, code ErrCode, msg string, metaPairs ...interface{}) err
 func Convert(err error) error {
 	if err == nil {
 		return nil
-	} else if e, ok := err.(*Error); ok {
+	}
+
+	var e *Error
+	if errors.As(err, &e) {
```

Comment:
I don't think it makes sense to use `errors.As` here; we don't want to return an inner error as that would drop any outer errors that wrap it.

Reply:
Yeah I guess it's fine to copy the fields (metadata, code, etc) from the inner error if it exists, while still preserving the full error chain

Reply:
Note I would still want to keep the original special case of the type assertion, as a fast path.